File: crates/ariadne-parser/fixtures/python/sample.py
import json
from pathlib import Path


class Counter:
    def __init__(self, start: int = 0) -> None:
        self.value = start

    def increment(self) -> int:
        self.value += 1
        return self.value


def make_counter(start: int) -> Counter:
    return Counter(start)


def load_config(path: str) -> dict:
    text = Path(path).read_text()
    return json.loads(text)

File: crates/ariadne-parser/fixtures/javascript/sample.js
import { readFile } from "node:fs/promises";

export class Counter {
  constructor(start = 0) {
    this.value = start;
  }

  increment() {
    this.value += 1;
    return this.value;
  }
}

export function makeCounter(start) {
  const counter = new Counter(start);
  return () => counter.increment();
}

export async function loadConfig(path) {
  const text = await readFile(path, "utf8");
  return JSON.parse(text);
}
